text/0004-remove-unnecessary-allocator-usage.md

```diff
@@ -0,0 +1,326 @@
+
+The `ext_input_size_version_1` host function returns the size in bytes of the input data.
+
+The `ext_input_read_version_1` host function copies some data from the input data to the memory of the runtime. The `offset` parameter indicates the offset within the input data where to start copying, and must be inferior or equal to the value returned by `ext_input_size_version_1`. The `out` parameter is [a pointer-size](https://spec.polkadot.network/chap-host-api#defn-runtime-pointer-size) containing the buffer where to write to.
+The runtime execution stops with an error if `offset` is strictly superior to the size of the input data, or if `out` is outside of the range of the memory of the virtual machine, even if the amount of data to copy would be 0 bytes.
+
+### Other changes
+
+In addition to the new host functions, this RFC proposes two changes to the runtime-host interface:
+
+- The following function signature is now also accepted for runtime entry points: `(func (result i64))`.
+- Runtimes no longer need to expose a constant named `__heap_base`.
+
+All the host functions that are being superceded by new host functions are now considered deprecated and should no longer be used.
+The following other host functions are similarly also considered deprecated:
+
+- `ext_storage_get_version_1`
+- `ext_default_child_storage_get_version_1`
+- `ext_allocator_malloc_version_1`
+- `ext_allocator_free_version_1`
+- `ext_offchain_network_state_version_1`
+
+## Drawbacks
+
+This RFC might be difficult to implement in Substrate due to the internal code design. It is not clear to the author of this RFC how difficult it would be.
+
+## Prior Art
+
+The API of these new functions was heavily inspired by API used by the C programming language.
+
+## Unresolved Questions
+
+The changes in this RFC would need to be benchmarked. This involves implementing the RFC and measuring the speed difference.
+
+It is expected that most host functions are faster or equal speed to their deprecated counterparts, with the following exceptions:
+
+- `ext_input_size_version_1`/`ext_input_read_version_1` is inherently slower than obtaining a buffer with the entire data due to the two extra function calls and the extra copying. However, given that this only happens once per runtime call, the cost is expected to be negligible.
+
+- The `ext_crypto_*_public_keys`, `ext_offchain_network_state`, and `ext_offchain_http_*` host functions are likely slightly slower than their deprecated counterparts, but given that they are used only in offchain workers this is acceptable.
+
+- It is unclear how replacing `ext_storage_get` with `ext_storage_read` and `ext_default_child_storage_get` with `ext_default_child_storage_read` will impact performances.
+
+- It is unclear how the changes to `ext_storage_next_key` and `ext_default_child_storage_next_key` will impact performances.
```

Comment:
For writing variable-length host functions, is it really necessary to revise it?
In fact, the host always needs an Allocator to support the copy of the entrypoint parameter.
So if the new version cannot improve performance, it will only increase the complexity of use.

Reply:
It's just what you are talking about:  `input to the runtime call`.